File: program/check-palindrome-number/CheckPalindromeNumber.js
export default (num) => {
    const length = num.toString().length;
    const digits = num.toString().split('');
    const realDigits = digits.map(Number);
    if (length % 2 === 0) {
        const firstPart = realDigits
            .filter((val, index) => index < realDigits.length / 2);
        const secondPartReversed = realDigits
            .filter((val, index) => index >= realDigits.length / 2)
            .reverse();
        const firstPartNumber = Number(firstPart
            .map((val) => val.toString())
            .join(''));
        const secondPartReversedNumber = Number(secondPartReversed
            .map((val) => val.toString())
            .join(''));
        return firstPartNumber === secondPartReversedNumber ? 'Palindrome Number' : 'Not Palindrome Number';
    } else {
        const firstPart = realDigits
            .filter((val, index) => index < Math.ceil(realDigits.length / 2));
        const secondPartReversed = realDigits
            .filter((val, index) => index >= Math.ceil(realDigits.length / 2 - 1))
            .reverse();
        const firstPartNumber = Number(firstPart
            .map((val) => val.toString())
            .join(''));
        const secondPartReversedNumber = Number(secondPartReversed
            .map((val) => val.toString())
            .join(''));
        return firstPartNumber === secondPartReversedNumber ? 'Palindrome Number' : 'Not Palindrome Number';
    }
};

File: program/check-palindrome-number/CheckPalindromeNumber.c
#include <stdio.h>
#include <stdbool.h>

bool isPalindrome(int x)
{
    static int rev = 0;
    static int num;
    num = x;
    if (x == 0)
    {
        return (num == rev);
    }
    rev = rev * 10 + x % 10;
    return isPalindrome(x / 10);
}
int main()
{
    int n;
    printf("Enter a number :");
    scanf("%d", &n);
    if (isPalindrome(n))
        printf("is a palindrome");
    else
        printf("is Not a palindrome");
    return 0;
}